sdk/helper/ldaputil/client.go

```diff
@@ -553,7 +558,7 @@ func (c *Client) GetLdapGroups(cfg *ConfigEntry, conn Connection, userDN string,
 	if cfg.UseTokenGroups {
 		entries, err = c.performLdapTokenGroupsSearch(cfg, conn, userDN)
 	} else {
-		if paging, ok := conn.(PagingConnection); ok && cfg.MaximumPageSize >= 0 {
+		if paging, ok := conn.(PagingConnection); ok && *cfg.MaximumPageSize >= 0 {
```

Comment:
I propose that if the page size is set to 0, a non-paged search should be done, as 0 is an invalid value to use as a page size.

```suggestion
		if paging, ok := conn.(PagingConnection); ok && *cfg.MaximumPageSize > 0 {
```